src/program/packetblaster/synth/synth.lua

```diff
@@ -0,0 +1,96 @@
+-- Use of this source code is governed by the Apache 2.0 license; see COPYING.
+
+module(..., package.seeall)
+
+local engine    = require("core.app")
+local config    = require("core.config")
+local timer     = require("core.timer")
+local pci       = require("lib.hardware.pci")
+local intel10g  = require("apps.intel.intel10g")
+local intel_app = require("apps.intel.intel_app")
+local basic_apps = require("apps.basic.basic_apps")
+local main      = require("core.main")
+local Synth     = require("apps.test.synth").Synth
+local LoadGen   = require("apps.intel.loadgen").LoadGen
+local lib = require("core.lib")
+local ffi = require("ffi")
+local C = ffi.C
+
+local usage = require("program.packetblaster.synth.README_inc")
+
+local long_opts = {
+   duration     = "D",
+   help         = "h",
+   src          = "s",
+   dst          = "d",
+   sizes        = "S"
+}
+
+function run (args)
+   local opt = {}
+   local duration
+   local c = config.new()
+   function opt.D (arg) 
+      duration = assert(tonumber(arg), "duration is not a number!")  
+   end
+   function opt.h (arg)
+      print(usage)
+      main.exit(1)
+   end
+
+   local source
+   local destination
+   local sizes
+   function opt.s (arg) source = arg end
+   function opt.d (arg) destination = arg end
+   function opt.S (arg)
+     sizes = {}
+     for size in string.gmatch(arg, "%d+") do
+       sizes[#sizes+1] = tonumber(size)
+     end
+   end
+
+   args = lib.dogetopt(args, opt, "hD:s:d:S:", long_opts)
+   config.app(c, "source", Synth, { sizes = sizes,
+   src = source,
```

Comment:
Yeah, my vim macro to align/autoindent lua has issues with functions across lines ;-)
Fixed it manually.